src/control/actor/host/notify.go

```diff
@@ -57,27 +53,41 @@ func (c *HostNotifyCmd) listenCloseF(net network.Network, addr ma.Multiaddr) {
 
 func (c *HostNotifyCmd) connectedF(net network.Network, conn network.Conn) {
 	logrus.Info("connection detected: ", conn.RemotePeer().String())
-
-	// request metadata as soon as we connect to a peer
-	peerData, err := PollPeerMetadata(conn.RemotePeer(), c.Base, c.PeerMetadataState, c.Store, c.PeerStore)
+	h, _ := c.Host()
+	// Request the Host Metadata
+	hInfo := ReqHostInfo(context.Background(), h, conn)
+	// Request the BeaconMetadata
+	bMetadata, err := ReqBeaconMetadata(context.Background(), h, conn.RemotePeer())
```

Comment:
i think this error should be handled in some way. if it errors you will `UpdateBeaconMetadata` in line 82 with some wrong data

Reply:
In this case, ReqBeaconMetadata will always provide an empty struct of the Beacon.MetaData. 
In case of not receiving it, I thought that a warning log was more than enough since we wanted to implement a data version struct in the future.

Do you think that just fetching the info to the Peer struct should be only done if the `err != nil` ?

Perhaps, so far, we would just like to update the Peer info if the request was successful, what do you think?

Reply:
What I mean is that if you don't check that `ReqBeaconMetadata` gave an error, you will end up calling `UpdateBeaconMetadata` with an empty `bMetadata`. Is that what you want?
```
peer.UpdateBeaconMetadata(bMetadata)
```

So what I'm suggesting is doing:
```
if err != nil {
  peer.UpdateBeaconMetadata(bMetadata)
}
```

Reply:
Updated in latest version